presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PartialAggregationPushDown.java

```diff
@@ -90,7 +90,7 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Void> cont
                 return context.defaultRewrite(node);
             }
 
-            boolean decomposable = node.isDecomposable(functionRegistry);
+            boolean decomposable = node.isDecomposable(functionRegistry) && node.getOrderBySymbols().isEmpty();
```

Comment:
Because ORDER BY only works when aggregation runs in SINGLE step, it does not work for ROLLUP and CUBE, but GROUPING SETS without empty grouping set would work. I added tests.